Sources/Commands/UserToolchain.swift

```diff
@@ -79,6 +77,19 @@ public struct UserToolchain: Toolchain {
                 searchPaths: envSearchPaths)
         }
 
+        let binDir: AbsolutePath
+        if let binDirOverride = binDirOverride {
+            binDir = binDirOverride
```

Comment:
Instead of override, I think the best thing to do is, store the binDir, instead of toolchain, in the destination.